src/ToricVarieties/ToricLineBundles/attributes.jl

```diff
@@ -72,12 +149,48 @@ export basis_of_global_sections_via_rational_functions
 @doc Markdown.doc"""
     basis_of_global_sections_via_homogeneous_component(l::ToricLineBundle)
 
-Return a basis of the global sections of the toric line bundle `l`,
-which is computed from a homogeneous component of the cox ring.
+Return a basis of the global sections of the toric line bundle `l`
+in terms of a homogeneous component of the Cox ring of `toric_variety(l)`.
+For convenience, this method can also be called via
+`basis_of_global_sections(l::ToricLineBundle)`.
+
+# Examples
+```jldoctest
+julia> v = projective_space(NormalToricVariety, 2)
+A normal, non-affine, smooth, projective, gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> l = ToricLineBundle(v, [fmpz(2)])
+A toric line bundle on a normal toric variety
+
+julia> basis_of_global_sections_via_homogeneous_component(l)
+6-element Vector{MPolyElem_dec{fmpq, fmpq_mpoly}}:
+ x3^2
+ x2*x3
+ x2^2
+ x1*x3
+ x1*x2
+ x1^2
+
+julia> basis_of_global_sections(l)
+6-element Vector{MPolyElem_dec{fmpq, fmpq_mpoly}}:
+ x3^2
+ x2*x3
+ x2^2
+ x1*x3
+ x1*x2
+ x1^2
+```
 """
 @attr function basis_of_global_sections_via_homogeneous_component(l::ToricLineBundle)
+    if has_attribute(toric_variety(l), :vanishing_sets)
+        tvs = vanishing_sets(toric_variety(l))[1]
+        if contains(tvs, l)
+            return []
```

Comment:
Same as above

Reply:
Added to https://github.com/oscar-system/Oscar.jl/issues/1257. I hope to fix this soon in a separate PR.